src/transformers/models/levit/modeling_levit.py

```diff
@@ -0,0 +1,729 @@
+
+        self.embedding_layer_2 = LevitConvEmbeddings(
+            config.embed_dim[0] // 8, config.embed_dim[0] // 4, config.kernel_size, config.stride, config.padding
+        )
+        self.activation_layer_2 = nn.Hardswish()
+
+        self.embedding_layer_3 = LevitConvEmbeddings(
+            config.embed_dim[0] // 4, config.embed_dim[0] // 2, config.kernel_size, config.stride, config.padding
+        )
+        self.activation_layer_3 = nn.Hardswish()
+
+        self.embedding_layer_4 = LevitConvEmbeddings(
+            config.embed_dim[0] // 2, config.embed_dim[0], config.kernel_size, config.stride, config.padding
+        )
+
+    def forward(self, pixel_values):
+        embeddings = self.embedding_layer_1(pixel_values)
+        embeddings = self.activation_layer_1(embeddings)
+        embeddings = self.embedding_layer_2(embeddings)
+        embeddings = self.activation_layer_2(embeddings)
+        embeddings = self.embedding_layer_3(embeddings)
+        embeddings = self.activation_layer_3(embeddings)
+        embeddings = self.embedding_layer_4(embeddings)
+        return embeddings
+
+
+class MLPLayerWithBN(nn.Module):
+    """
+    MLP Layer with Batch Norm
+    """
+
+    def __init__(self, input_dim, output_dim, bn_weight_init=1):
+        super().__init__()
+        self.linear = nn.Linear(in_features=input_dim, out_features=output_dim, bias=False)
+        self.bn = nn.BatchNorm1d(output_dim)
+
+    def forward(self, hidden_state):
+        hidden_state = self.linear(hidden_state)
+        hidden_state = self.bn(hidden_state.flatten(0, 1)).reshape_as(hidden_state)
+        return hidden_state
+
+
+class LevitSubsample(nn.Module):
+    """
+    Subsampling Module
+    """
+
+    def __init__(self, stride, resolution):
+        super().__init__()
+        self.stride = stride
+        self.resolution = resolution
+
+    def forward(self, hidden_state):
+        batch_size, N, channels = hidden_state.shape
+        hidden_state = hidden_state.view(batch_size, self.resolution, self.resolution, channels)[
+            :, :: self.stride, :: self.stride
+        ].reshape(batch_size, -1, channels)
+        return hidden_state
+
+
+class LevitAttention(nn.Module):
+    """
+    LeViT Attention Module
+    """
+
+    def __init__(self, embed_dim, key_dim, num_heads, attention_ratio, resolution):
+        super().__init__()
+        self.num_heads = num_heads
+        self.scale = key_dim**-0.5
+        self.key_dim = key_dim
+        self.attention_ratio = attention_ratio
+        self.out_dim_qkv = attention_ratio * key_dim * num_heads + key_dim * num_heads * 2
+        self.out_dim_proj = attention_ratio * key_dim * num_heads
+
+        self.qkv = MLPLayerWithBN(embed_dim, self.out_dim_qkv)
+        self.activation = nn.Hardswish()
+        self.projection = MLPLayerWithBN(self.out_dim_proj, embed_dim, bn_weight_init=0)
+
+        points = list(itertools.product(range(resolution), range(resolution)))
+        N = len(points)
```

Comment:
We don't really like one-letter variable names, so could you rename this to something more clear?

Reply:
Done